### PR TITLE
[OapExtensions] Inject OapStrategies to SparkSessionExtensions #999

### PR DESCRIPTION
**Pull request by LuciferYang**
## What changes were proposed in this pull request?

Follow-up #995, Inject `OapStrategies` to `OapExtensions` instead of add strategies to `OapSessionStateBuilder`, now we no longer need `OapSession` & `OapSessionStateBuilder`, use SparkSession and config `spark.sql.extensions` to  `OapExtensions` is enough.

There will another follow-up to clean up codes that originally created `OapSession` like OapEnv.init().

## How was this patch tested?

`OapPlannerSuite` cover this refactor and rename it to `OapStrategySuite`
 Add `OapExtensionsSuite` to test `OapExtensions` contents.

### CODE REVIEW

File: src/test/scala/org/apache/spark/sql/TestOap.scala [hunk 1]
```diff
@@ -18,14 +18,14 @@ package org.apache.spark.sql
 
 import org.apache.spark.SparkConf
 import org.apache.spark.sql.internal.oap.OapConf
-import org.apache.spark.sql.oap.OapSession
 
 object TestOap extends TestOapContext(
-  OapSession.builder.config(
+  SparkSession.builder.config(
```
Comment (LuciferYang) [inline src/test/scala/org/apache/spark/sql/TestOap.scala]: Maybe we can remove this class and `OapStrategySuite` should extends `SharedOapContext`.

Comment (shaowenzh) [reply src/test/scala/org/apache/spark/sql/TestOap.scala]: Hi Lucifer, thanks for work, I like this idea. Also I checked the source code, they have the test code like this:
```
test("inject spark planner strategy") {
    withSession(_.injectPlannerStrategy(MySparkStrategy)) { session =>
      assert(session.sessionState.planner.strategies.contains(MySparkStrategy(session)))
    }
  }
```
I guess maybe we can add something like this here in the future to verify the strategy has been added? Here is the link for reference: https://github.com/apache/spark/blob/v2.4.0-rc5/sql/core/src/test/scala/org/apache/spark/sql/SparkSessionExtensionSuite.scala
I haven't read through the whole OAP code.  Feel free to correct me if this kind of test is already there.

Comment (LuciferYang) [reply src/test/scala/org/apache/spark/sql/TestOap.scala]: @shaowenzh Added OapExtensionsSuite to test this.

File: src/main/scala/org/apache/spark/sql/execution/datasources/oap/OapStrategies.scala [hunk 2]
```diff
@@ -40,254 +40,7 @@ import org.apache.spark.sql.internal.oap.OapConf
...
-            case Some(fastScan) => OapAggregationFileScanExec(aggExpressions, projectList, fastScan)
-            case _ => PlanLater(child)
-          }
-        } else {
-          PlanLater(child)
-        }
-      case _ => PlanLater(child)
-    }
-  }
+trait OapStrategy extends Logging {
```
Comment (LinhongLiu) [inline src/main/scala/org/apache/spark/sql/execution/datasources/oap/OapStrategies.scala]: OapStrategy extends Strategy?

Comment (LuciferYang) [reply src/main/scala/org/apache/spark/sql/execution/datasources/oap/OapStrategies.scala]: got it ~

Comment (LuciferYang) [reply src/main/scala/org/apache/spark/sql/execution/datasources/oap/OapStrategies.scala]: done ~

**Comment by songzhan01** [comment]
lgtm, good work.

**Pull request merged**